src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestClientInternal.java

```diff
@@ -170,20 +170,34 @@ public class SnowflakeStreamingIngestClientInternal<T> implements SnowflakeStrea
     if (!isTestMode) {
       // Setup request builder for communication with the server side
       this.role = prop.getProperty(Constants.ROLE);
-      try {
-        KeyPair keyPair =
-            Utils.createKeyPairFromPrivateKey(
-                (PrivateKey) prop.get(SFSessionProperty.PRIVATE_KEY.getPropertyKey()));
-        this.requestBuilder =
-            new RequestBuilder(
-                accountURL,
-                prop.get(USER).toString(),
-                keyPair,
-                this.httpClient,
-                String.format("%s_%s", this.name, System.currentTimeMillis()));
-      } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
-        throw new SFException(e, ErrorCode.KEYPAIR_CREATION_FAILURE);
+
+      Object credential = null;
+
+      // Authorization type will be set to jwt by default
+      if (prop.getProperty(Constants.AUTHORIZATION_TYPE).equals(Constants.JWT)) {
+        try {
+          credential =
+              Utils.createKeyPairFromPrivateKey(
+                  (PrivateKey) prop.get(SFSessionProperty.PRIVATE_KEY.getPropertyKey()));
+        } catch (NoSuchAlgorithmException | InvalidKeySpecException e) {
+          throw new SFException(e, ErrorCode.KEYPAIR_CREATION_FAILURE);
+        }
+        logger.logInfo("Using JWT KeyPair for authorization");
```

Comment:
nit: could we combine the log line with the one below and use the authType as an input?

Reply:
Fixed.